dali/kernels/slice/slice_gpu.cuh

```diff
@@ -72,9 +72,28 @@ struct SliceBlockDesc {
   uint64_t size;
 };
 
-template<typename OutputType>
-constexpr int coalesced_values = sizeof(OutputType) >= 4 ? 1 : 4 / sizeof(OutputType);
-
+template<typename T>
+union PackedBuffer {
+  using PackedType = uint32_t;
+  static constexpr size_t kCapacity = sizeof(T) >= sizeof(PackedType) ?
+                                      1 : sizeof(PackedType) / sizeof(T);
+
+  T values[kCapacity];
+  PackedType raw;
+
+  __device__ inline void store(T* mem, size_t count) {
+    if (kCapacity == 1) {
+      *mem = *values;
+    } else if (count == kCapacity && (uint64_t)mem % sizeof(PackedType) == 0) {
+      *reinterpret_cast<PackedType*>(mem) = raw;
+    } else {
+      #pragma unroll
+      for (size_t i = 0; i < count; i++) {
+        mem[i] = values[i];
```

Comment:
Same as above.

Reply:
Hmm, I think this is valid code and not UB, because both `mem` and `values` are arrays of `T`. Am I missing something?

Reply:
My bad.